pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -218,11 +225,16 @@ public static class Broker {
     public static final String CONFIG_OF_BROKER_GROUPBY_TRIM_THRESHOLD = "pinot.broker.groupby.trim.threshold";
     public static final int DEFAULT_BROKER_GROUPBY_TRIM_THRESHOLD = 1_000_000;
 
+    // Configure the request handler type used by broker to handler inbound query request.
+    // NOTE: the request handler type refers to the communication between Broker and Server.
+    public static final String BROKER_REQUEST_HANDLER_TYPE_JSON_OVERRIDE_KEY = "handler.type";
```

Comment:
(nit) Better place for this is the Request specific options under the Broker options

See this - https://github.com/apache/pinot/blob/master/pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java#L246

Also, can we name it something like `"use.multistage.engine"` or something similar